### PR TITLE
Implement serialization logic for ProjectorSums.

### PR DESCRIPTION
**Pull request by tonybruguier**
This is step 2 of issue #357

### CODE REVIEW

**Comment by tonybruguier** [comment]
@MichaelBroughton 
Thanks for merging #621. The present PR can now be reviewed, I think and if you have the time, because its unit tests now pass.

**Comment by tonybruguier** [comment]
Friendly ping. I could work on this over the Thanksgiving break if you have more comments.

File: tensorflow_quantum/core/serialize/serializer.py [hunk 1]
```diff
@@ -970,3 +971,72 @@ def _process_pauli_type(char):
...
+        projectorsum = cirq.ProjectorSum.from_pauli_strings(projectorsum)
+
+    if not isinstance(projectorsum, cirq.ProjectorSum):
+        raise TypeError("serialize requires a cirq.ProjectorSum object."
+                        " Given: " + str(type(projectorsum)))
+
+    if any(not isinstance(qubit, cirq.GridQubit)
+           for qubit in projectorsum.qubits):
+        raise ValueError("Attempted to serialize a paulisum that doesn't use "
+                         "only cirq.GridQubits.")
```
Comment (MichaelBroughton) [inline tensorflow_quantum/core/serialize/serializer.py:995]: We can do LineQubits now too :)

File: tensorflow_quantum/core/serialize/serializer.py [hunk 2]
```diff
@@ -970,3 +971,72 @@ def _process_pauli_type(char):
...
+        projectorterm_proto.coefficient_imag = term.coefficient.imag
+        for qubit, basis_state in sorted(
+                term.projector_dict.items()):  # sort to keep qubits ordered
+            if basis_state:
+                projectorterm_proto.projector_dict.add(
+                    qubit_id=op_serializer.qubit_to_proto(qubit),
+                    basis_state=basis_state)
+            else:
+                projectorterm_proto.projector_dict.add(
+                    qubit_id=op_serializer.qubit_to_proto(qubit))
```
Comment (MichaelBroughton) [inline tensorflow_quantum/core/serialize/serializer.py:1011]: Nit: Could we not do:
```python3
projectorterm_proto.projector_dict.add(
                    qubit_id=op_serializer.qubit_to_proto(qubit),
                    basis_state=basis_state)
```
In the case of basis_state being true or false ?

File: tensorflow_quantum/core/serialize/serializer_test.py [hunk 3]
```diff
@@ -407,6 +408,29 @@ def _get_valid_pauli_proto_pairs():
     return pairs
 
 
+def _get_valid_projector_proto_pairs():
+    """Generate valid projectorsum proto pairs."""
+    q0 = cirq.GridQubit(0, 0)
+    q1 = cirq.GridQubit(1, 0)
```
Comment (MichaelBroughton) [inline tensorflow_quantum/core/serialize/serializer_test.py:414]: Maybe throw in some `LineQubit`s here too ?

**Review by MichaelBroughton — DISMISSED** [review]
LGTM after some nits. This is great, sorry again for the slow review.

File: tensorflow_quantum/core/serialize/serializer.py (hunk 1 above)
Comment (tonybruguier) [reply tensorflow_quantum/core/serialize/serializer.py:995]: Done.

File: tensorflow_quantum/core/serialize/serializer.py (hunk 2 above)
Comment (tonybruguier) [reply tensorflow_quantum/core/serialize/serializer.py:1011]: Done.

File: tensorflow_quantum/core/serialize/serializer_test.py (hunk 3 above)
Comment (tonybruguier) [reply tensorflow_quantum/core/serialize/serializer_test.py:414]: Done.

**Review by MichaelBroughton — APPROVED** [review]

**Pull request merged**